disco_aws_automation/disco_autoscale.py

```diff
@@ -117,32 +120,26 @@ def clean_configs(self):
             except BotoServerError:
                 pass
 
-    def delete_group(self, hostclass, force=False):
+    def delete_groups(self, hostclass=None, group_name=None, force=False):
         '''Delete a specific Autoscaling Group'''
-        throttled_call(self.connection.delete_auto_scaling_group,
-                       self.get_groupname(hostclass), force_delete=force)
-
-    def clean_groups(self, force=False):
-        '''Delete unused Autoscaling Groups in current environment'''
-        for group in self._filter_by_environment(self.get_groups()):
+        groups = self.get_existing_groups(hostclass=hostclass, group_name=group_name)
+        for group in groups:
             try:
-                self.delete_group(self.get_hostclass(group.name), force)
+                throttled_call(group.delete, force_delete=force)
+                logging.info("Deleting group %s", group.name)
+                self.delete_config(group.launch_config_name)
             except BotoServerError:
-                pass
+                logging.info("Unable to delete group %s, try force deleting", group.name)
 
-    def scaledown_group(self, hostclass):
-        '''Scales down number of instances in Autoscaling group to zero'''
-        group_list = throttled_call(self.connection.get_all_groups,
-                                    names=[self.get_groupname(hostclass)])
-        if group_list:
-            group = group_list[0]
-            group.min_size = group.max_size = group.desired_capacity = 0
-            throttled_call(group.update)
+    def clean_groups(self, force=False):
+        '''Delete unused Autoscaling Groups in current environment'''
+        self.delete_groups(force=force)
```

Comment:
Depends on how it's called. If force isn't specified then it won't delete an in-use ASG. Confusing, but that's how it worked before I touched it.

I will update the docstring to be clearer though.